db/hybrid_logical_vector_test.go

```diff
@@ -229,3 +229,66 @@ func TestHybridLogicalVectorPersistence(t *testing.T) {
 	assert.Equal(t, inMemoryHLV.PreviousVersions, hlvFromPersistance.PreviousVersions)
 	assert.Equal(t, inMemoryHLV.MergeVersions, hlvFromPersistance.MergeVersions)
 }
+
+func TestAddNewerVersionsBetweenTwoVectorsWhenNotInConflict(t *testing.T) {
+	testCases := []struct {
+		name          string
+		localInput    []string
+		incomingInput []string
+		expected      []string
+	}{
+		{
+			name:          "testcase1",
+			localInput:    []string{"abc@15"},
+			incomingInput: []string{"def@25", "abc@20"},
+			expected:      []string{"def@25", "abc@20"},
+		},
+		{
+			name:          "testcase2",
+			localInput:    []string{"abc@15", "def@30"},
+			incomingInput: []string{"def@35", "abc@15"},
+			expected:      []string{"def@35", "abc@15"},
+		},
+		{
+			name:          "testcase3",
```

Comment:
This test case is also a conflict (incoming HLV doesn't dominate the local cv), so I think should be omitted from this test.

Reply:
Okay I agree I didn't notice that, I have removed this case. Does the function IsInConflict need changing then? When running localHLV.IsInConflict(incomingHLV) no conflict is detected because the function checks if either one is dominating over the other. Should this be changed to only check one way?